test/OpenTelemetry.AutoInstrumentation.Tests/Diagnostics/SdkSelfDiagnosticsEventListenerTests.cs

```diff
@@ -16,57 +16,75 @@
 
 // Source originated from https://github.com/open-telemetry/opentelemetry-dotnet/blob/23609730ddd73c860553de847e67c9b2226cff94/test/OpenTelemetry.Tests/Internal/SelfDiagnosticsEventListenerTest.cs
 
-using System;
+using System.Collections.Generic;
 using System.Diagnostics.Tracing;
-using System.IO;
 using FluentAssertions;
 using OpenTelemetry.AutoInstrumentation.Diagnostics;
+using OpenTelemetry.AutoInstrumentation.Logging;
 using Xunit;
 
+namespace OpenTelemetry.AutoInstrumentation.Tests.Diagnostics;
+
 public class SdkSelfDiagnosticsEventListenerTests
 {
     [Fact]
     public void EventSourceSetup_LowerSeverity()
     {
-        var originalConsoleOut = Console.Out; // preserve the original stream
-        using var writer = new StringWriter();
-        var listener = new SdkSelfDiagnosticsEventListener(EventLevel.Error);
-
-        // Emitting a Verbose event. Or any EventSource event with lower severity than Error.
-        AspNetTelemetryEventSource.Log.ActivityRestored("123");
-        OpenTelemetrySdkEventSource.Log.ActivityStarted("Activity started", "1");
-
-        // Prepare the output for assertion
-        writer.Flush();
-        var outputString = writer.GetStringBuilder().ToString();
-        outputString.Should().NotContain("EventSource=OpenTelemetry-Instrumentation-AspNet-Telemetry, Message=Activity restored, Id='123'");
-        outputString.Should().NotContain("EventSource=OpenTelemetry-Sdk, Message=Activity started.");
-
-        // Cleanup
-        Console.SetOut(originalConsoleOut);
+        // get default sink using reflection
+        var defaultLogger = OtelLogging.GetLogger();
+        var sinkField = defaultLogger.GetType().GetField("_sink", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
+        var defaultSink = (ISink)sinkField.GetValue(defaultLogger);
+
+        try
+        {
+            var testSink = new TestSink();
+            var listener = new SdkSelfDiagnosticsEventListener(EventLevel.Error);
+
+            // override default logger's sink with test sink
+            sinkField.SetValue(defaultLogger, testSink);
```

Comment:
I considered other options, but wanted to stay consistent how logger is obtained (didn't want to inject it into ctor of `SdkSelfDiagnosticEventListener` when other classes obtain it directly from `OtelLogging`), didn't want to add methods to `Logger` or `SdkSelfDiagnosticEventListener` for the sake of these tests. Let me know if find this approach acceptable.